agent/Cargo.toml

```diff
@@ -17,8 +17,8 @@ tracing-subscriber = { version = "0.3", features = ["registry", "env-filter"] }
 tracing-opentelemetry = "0.18"
 
 # k8s-openapi must match the version required by kube and enable a k8s version feature
-k8s-openapi = { version = "0.17.0", default-features = false, features = ["v1_20"] }
-kube = { version = "0.79.0", default-features = false, features = [ "derive", "runtime", "rustls-tls" ] }
+k8s-openapi = { version = "0.18.0", default-features = false, features = ["v1_24"] }
+kube = { version = "0.82.2", default-features = false, features = [ "derive", "runtime", "rustls-tls" ] }
```

Comment:
Can we change this to `0.82` so we can consume patch releases? I've been meaning to get to it so that all our dependencies look like that

Might be worth doing that in this PR too?